packages/mdc-base/foundation.ts

```diff
@@ -21,44 +21,35 @@
  * THE SOFTWARE.
  */
 
-/**
- * @template A
- */
-class MDCFoundation {
-  /** @return enum{cssClasses} */
-  static get cssClasses() {
+class MDCFoundation<AdapterType extends {} = {}> {
+  static get cssClasses(): {} {
     // Classes extending MDCFoundation should implement this method to return an object which exports every
     // CSS class the foundation class needs as a property. e.g. {ACTIVE: 'mdc-component--active'}
     return {};
   }
 
-  /** @return enum{strings} */
-  static get strings() {
+  static get strings(): {} {
     // Classes extending MDCFoundation should implement this method to return an object which exports all
     // semantic strings as constants. e.g. {ARIA_ROLE: 'tablist'}
     return {};
   }
 
-  /** @return enum{numbers} */
-  static get numbers() {
+  static get numbers(): {} {
     // Classes extending MDCFoundation should implement this method to return an object which exports all
     // of its semantic numbers as constants. e.g. {ANIMATION_DELAY_MS: 350}
     return {};
   }
 
-  /** @return {!Object} */
-  static get defaultAdapter() {
+  static get defaultAdapter(): {} {
```

Comment:
Should this be typed as AdapterType? (I might be wrong here but it stands out to me)

Reply:
Since defaultAdapter is static, there is no way to know what AdapterType is before an instance of MDCFoundation is created.

https://github.com/Microsoft/TypeScript/issues/24018